lib/webinspector/transformer/webinspector-socket.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable promise/prefer-await-to-callbacks */
+import Stream from 'stream';
+import _ from 'lodash';
+
+
+const CHUNK_SIZE = 5000;
+
+class WebInspectorSocket extends Stream.Transform {
+  constructor (socketClient, opts = {}) {
+    super({ objectMode: true });
+
+    this._socketClient = socketClient;
+    this._socketClient.setNoDelay(true);
+    this._socketClient.setKeepAlive(true);
+
+    this._socketChunkSize = opts.socketChunkSize || CHUNK_SIZE;
+  }
+
+  _transform (data, encoding, callback) {
+    this._socketClient.cork();
+    if (_.isBuffer(data) || _.isString(data)) {
+      let start = 0, end = this._socketChunkSize;
+      while (start < data.length) {
+        const chunkBuffer = data.slice(start, end);
+        start = end;
+        end = end + this._socketChunkSize;
+        this._socketClient.write(chunkBuffer, encoding);
```

Comment:
Are you sure this does the trick? Since the window size in tcp happens on the socket layer, I feel like it shouldn't work

Reply:
It works for me, on two different devices, and for a few folks who tested things

Reply:
The size might need to be tuned for performance, which is why it is configurable.